core/src/main/java/org/elasticsearch/search/suggest/phrase/PhraseSuggestionBuilder.java

```diff
@@ -417,7 +418,7 @@ public Script collateQuery() {
      */
     public PhraseSuggestionBuilder collateParams(Map<String, Object> collateParams) {
         Objects.requireNonNull(collateParams, "collate parameters cannot be null.");
-        this.collateParams = new HashMap<>(collateParams);
+        this.collateParams = new LinkedHashMap<>(collateParams);
```

Comment:
I agree, I have created a new `StreamOut` `writeMapWithConsistentOrder`, when call this method to write a stream, it will keep a consistent order for maps.